### ISSUE TITLE
Not all rows are returned when encountering an empty string in the first column

### ISSUE THREAD

**Issue by NeKzor** [denodrivers/mysql#162]
What a waste of time this was to debug just to find out that this library cannot handle simple select queries. The current workaround is to select another column that cannot contain an empty string before the column that could contain an empty string. This is a big problem when you select large amounts of data from a table and nobody would notice that the data is incomplete! I actually wonder now where the issue is...

Minimal reproducible example:

```ts
const rows = await db.query(`select '' as a`);
console.log(rows);
```

Actual output:

```ts
INFO connecting 127.0.0.1:3307
INFO connected to 127.0.0.1:3307
[]
INFO close connection
```

Here is a working example with `npm:mysql2/promise`:

```ts
const [rows] = await db.query(`select '' as a`);
console.log(rows);
```

Expected output:

```ts
[ { a: "" } ]
```

Used Deno version:

```
deno 1.36.4 (release, x86_64-unknown-linux-gnu)
v8 11.6.189.12
typescript 5.1.6
```

Used database images from Docker Hub:
- mariadb:11.1
- mariadb:11
- mariadb:10
- mysql:8
- mysql:5

**Comment by NeKzor** [comment]
Oh wow, it looks like this is a regression caused by e3359d0465b5d59e6c11c5629dcea54456b4b7b8 (PR #140).
CC @shiyuhang0 @lideming

**Comment by lideming** [comment]
@NeKzor Thanks for reporting this.
> it looks like this is a regression caused by https://github.com/denodrivers/mysql/commit/e3359d0465b5d59e6c11c5629dcea54456b4b7b8 (PR https://github.com/denodrivers/mysql/pull/140).

You are right. The packet header byte is always `0xfe` whether it's an EOF_Packet or an OK_Packet representing EOF.

**Comment by NeKzor** [comment]
Thanks for the really fast fix! It's working again in v2.12.1.

**Comment by shiyuhang0** [comment]
Sorry for the mistake. I think mysql/mariadb may not following the https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_basic_ok_packet.html.
I also see the replacement of eof is not that easy in https://github.com/go-sql-driver/mysql/pull/1153
